engine/src/test/java/io/camunda/zeebe/engine/state/migration/to_8_3/MultiTenancyMigrationTest.java

```diff
@@ -148,6 +148,46 @@ void shouldMigrateProcessByIdAndVersionColumnFamily() {
       assertThat(legacyState.getProcessByProcessIdAndVersion(wrapString("processId"), 1)).isNull();
     }
 
+    @Test
+    void shouldMigrateProcessByIdAndVersionColumnFamilyUsingVersionManager() {
+      // given
+      final var model = Bpmn.createExecutableProcess("processId").startEvent().done();
+      legacyState.putProcess(
+          123,
+          new ProcessRecord()
+              .setKey(123)
+              .setBpmnProcessId("processId")
+              .setVersion(1)
+              .setResourceName("resourceName")
+              .setResource(wrapString(Bpmn.convertToString(model)))
+              .setChecksum(wrapString("checksum")));
+      // the version manager must be migrated first to ensure that the known versions are set
+      new ProcessDefinitionVersionMigration().runMigration(processingState);
```

Comment:
To be fair, this is only because of where you copied the legacy state from. If you'd copied it from a more recent version you wouldn't need to do this 😄 

I'm perfectly fine with this though.

Reply:
😆 Ah lol, okay. I feel that the state classes are not really set up for easy migration. You're idea for #13813 would help make them easier to migrate. I discussed this in #14468. Perhaps this could also help simplify such inter-dependencies between migration tasks.